File: Network.Angular/apps/Network.Angular/src/app/_models/shared/Comment.ts
import {User} from "../users/User";

export class Comment {

  constructor(
    public id: number | undefined,
    public dataPublication: Date,
    public text: string | undefined,
    public postId: number | undefined,
    public userid: number | undefined,
    public user: User | undefined
  ) {}
}

File: Network.Domain/Models/Comment.cs

using System.Text.Json.Serialization;

namespace Network.Domain.Models;

public class Comment : BaseEntity
{
    public DateTime DatePublication { get; set; }
    public string? Text { get; set; } // до 500 символов
    
    public int PostId { get; set; }
    [JsonIgnore]
    public Post Post { get; set; }
    
    public int? UserId { get; set; }
    //[JsonIgnore]
    public User? User { get; set; }
}